### PR TITLE
ref: more Azure remote config clarifications

### LINKED ISSUE #2376: remote: 2 Qs about Azure params
> Continues https://github.com/iterative/dvc.org/pull/2226#issuecomment-806200723

1. What about `AZURE_STORAGE_CONNECTION_STRING` and `AZURE_STORAGE_CONTAINER_NAME`?
2. What happens first, Azure "default credential" or Microsoft account login?

### LINKED ISSUE #2406: remote add: review Azure section
From https://github.com/iterative/dvc.org/pull/2390#issuecomment-824519403:

> I'm not sure we communicate properly defaults, available options, etc.

Context: after updating and rewriting the Azure section in https://dvc.org/doc/command-reference/remote/modify#available-parameters-per-storage-type we noticed that this remote is pretty complicated, and it's not clear what the recommended/assumed auth method should be, and which others if any to include in `remote add`.

See also #1859.

### PR DESCRIPTION
**Pull request by jorgeorpinel**
> Follow up to #2390

Attempts to close #2376
Closes #2406

### CODE REVIEW

File: content/docs/command-reference/remote/add.md [hunk 1]
```diff
@@ -171,18 +171,15 @@ $ dvc remote modify --local myremote connection_string 'mysecret'
...
 ```dvc
 $ export AZURE_STORAGE_CONNECTION_STRING='mysecret'
 $ dvc remote add -d myremote azure://mycontainer/path
 ```
 
-> 💡 Make sure the value is quoted so its processed correctly by the console.
-
-See `dvc remote modify` for a full list of Azure parameters and environment
-variables.
+See `dvc remote modify` for a full list of Azure parameters and env vars for
```
Comment (isidentical) [inline content/docs/command-reference/remote/add.md:181]: I might be missing context on this. How does `dvc remote modify` shows options? 
```
 $ dvc remote modify origin       
ERROR: the following arguments are required: option
usage: dvc remote modify [-h] [--global | --system | --project | --local] [-q | -v] [-u] name option [value]

Modify the configuration of a data remote.
Documentation: <https://man.dvc.org/remote/modify>

positional arguments:
  name           Name of the remote
  option         Name of the option to modify.
  value          (optional) Value of the option.

optional arguments:
  -h, --help     show this help message and exit
  --global       Use global config.
  --system       Use system config.
  --project      Use project config (.dvc/config).
  --local        Use local config (.dvc/config.local).
  -q, --quiet    Be quiet.
  -v, --verbose  Be verbose.
  -u, --unset    Unset option.

```

Comment (jorgeorpinel) [reply content/docs/command-reference/remote/add.md:181]: Yeah those code blocks become links in the cmd ref:

image

So you click on it and it goes to the `remote modify` ref, where you can find the Azure section with all the auth details. That's the idea.

Comment (isidentical) [reply content/docs/command-reference/remote/add.md:181]: Ah, TIL.

Comment (jorgeorpinel) [reply content/docs/command-reference/remote/add.md:181]: Haha. So approved? 🙂

**Review by isidentical — APPROVED** [review]
Thanks, look great

**Review by shcheklein — APPROVED** [review]

**Pull request merged**